x/sync/g_db/db_client.go

```diff
@@ -16,6 +16,13 @@ import (
 	pb "github.com/ava-labs/avalanchego/proto/pb/sync"
 )
 
+func MaybeBytesToMaybe(mb *pb.MaybeBytes) merkledb.Maybe[[]byte] {
```

Comment:
nit: should we unexport `MaybeBytesToMaybe` in both packages if we are replicating this function or should we use one implementation?

Reply:
I feel like this function should be defined once with the rest of the maybe code

Reply:
It is more of a proto thing though, but you can't define it with protos either

Reply:
Not sure I follow. Can't we define this in `maybe.go`?

Reply:
we could, but it isn't really a maybe functionality thing, it is like a codec/ marshaling thing which lives in proto and codec.  It feels weird to have this in the merkledb module since it isn't called anywhere in the merkledb module.  There was also talk of moving maybe out of merkledb to someplace more common.

Reply:
This would be in line with the `UnmarshalProto` and `ToProto` definitions in `merkledb`, which also aren't used in `merkledb`.  I feel like it makes sense for an operation defined on `Maybe` to live with the maybe code. I agree that moving `maybe` to its own package would make sense in a subsequent PR.

Reply:
Also related to the above, I think it might make sense to name this `UnmarshalProto` for parallelism with other similar functions

Reply:
It isn't a Maybe function, it is a maybebytes function.  I'm just going to remove its existence for now.